src/ejabberd_gen_sm.erl

```diff
@@ -32,78 +35,76 @@
          get_sessions/4, create_session/5, update_session/5, delete_session/5, cleanup/2,
          total_count/1, unique_count/1]).
 
--ignore_xref([behaviour_info/1, cleanup/2]).
+-ignore_xref([cleanup/2, behaviour_info/1]).
 
--spec start(module(), list()) -> any().
-start(Mod, Opts) ->
-    Mod:start(Opts).
+-define(MAIN_MODULE, ejabberd_sm).
+-type host_type() :: mongooseim:host_type() | global.
```

Comment:
I'd call it `host_type_or_global` like in `mongoose_backend` so that it's less surprising, but this may be a matter of taste.